src/services/recovery/setup.ts

```diff
@@ -82,3 +88,117 @@ export function getRecoverySetup({
     transactions,
   }
 }
+
+export async function _getEditRecoveryTransactions({
+  newTxCooldown,
+  newTxExpiration,
+  newGuardians,
+  moduleAddress,
+  provider,
+}: {
+  newTxCooldown: string
+  newTxExpiration: string
+  newGuardians: Array<string>
+  moduleAddress: string
+  provider: Web3Provider
+}): Promise<Array<MetaTransactionData>> {
+  const delayModifierContract = getModuleInstance(KnownContracts.DELAY, moduleAddress, provider)
+
+  const [txExpiration, txCooldown, [guardians]] = await Promise.all([
+    delayModifierContract.txExpiration(),
+    delayModifierContract.txCooldown(),
+    delayModifierContract.getModulesPaginated(SENTINEL_ADDRESS, MAX_GUARDIAN_PAGE_SIZE),
+  ])
+
+  // Recovery management transaction data
+  const txData: Array<string> = []
+
+  // Update cooldown
+  if (!txCooldown.eq(newTxCooldown)) {
+    const setTxCooldown = delayModifierContract.interface.encodeFunctionData('setTxCooldown', [newTxCooldown])
+    txData.push(setTxCooldown)
+  }
+
+  // Update expiration
+  if (!txExpiration.eq(newTxExpiration)) {
+    const setTxExpiration = delayModifierContract.interface.encodeFunctionData('setTxExpiration', [newTxExpiration])
+    txData.push(setTxExpiration)
+  }
+
+  // Cache guardian changes to determine prevModule
+  let _guardians = [...guardians]
+
+  // Don't add/remove same owners
+  const guardiansToAdd = newGuardians.filter(
+    (newGuardian) => !_guardians.some((oldGuardian) => sameAddress(oldGuardian, newGuardian)),
+  )
+  const guardiansToRemove = _guardians.filter(
+    (oldGuardian) => !newGuardians.some((newGuardian) => sameAddress(newGuardian, oldGuardian)),
+  )
+
+  for (const guardianToAdd of guardiansToAdd) {
+    const enableModule = delayModifierContract.interface.encodeFunctionData('enableModule', [guardianToAdd])
+    txData.push(enableModule)
+
+    // Add guardian to cache
+    _guardians.push(guardianToAdd)
```

Comment:
Do we actually need to add these to the cache?
The guardiansToAdd addresses can not be part of the guardiansToRemove. So they will never be relevant for removing a guardian.

Reply:
True, I've removed it in 09cad6f8bc952f8764655756af37d80a1eb86109.